packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -66,10 +72,25 @@ export class AndroidEmulatorDevice extends DeviceBase {
     ]);
   }
 
-  async bootDevice() {
-    if (this.emulatorProcess) {
-      this.emulatorProcess.kill(9);
+  private async ensureOldEmulatorProcessExited() {
+    let runningPid: string | undefined;
+    const subprocess = exec("ps", ["-Ao", "pid,command"]);
+    const regexpPattern = new RegExp(`(\\d+)\\s.*qemu.*-avd ${this.avdId}`);
+    lineReader(subprocess).onLineRead(async (line) => {
+      const regExpResult = regexpPattern.exec(line);
+      if (regExpResult) {
+        runningPid = regExpResult[1];
+      }
+    });
+    await subprocess;
+    if (runningPid) {
+      await exec("kill", ["-9", `${runningPid}`]);
```

Comment:
process.kill(runningPid, 9)

watch out as this method may throw when PID doesn't exist. Also PID needs to be passed as number

Reply:
https://nodejs.org/api/process.html#processkillpid-signal